src/leaflike/validator.clj

```diff
@@ -3,7 +3,7 @@
 
 (defn is-id?
   [body]
-  (let [id (:id body)]
+  (let [id (Integer/parseInt (:id body))]
```

Comment:
`Integer/parseInt` can throw exceptions, which would make `is-id?` not a predicate anymore.